### DISCUSSION TITLE
Why is `VCPKG_APPLOCAL_DEPS` set to `OFF` in `vcpkg_cmake_configure.cmake`?

### DISCUSSION THREAD

**Discussion by F-I-D-O** [microsoft/vcpkg#39604] [Q&A]
I was struggling to find why the build of my package is failing during the vcpkg install while it works when building when running CMake directly. I've found out that the reason is that runtime dependencies are missing in the vcpkg buildtree dir. This is because the libraries are copied automatically by the vcpkg toolchain file, a functionality that is disabled when building the port (option `VCPKG_APPLOCAL_DEPS`) 

I could definitely hack it somehow, but I feel like I'm missing the philosophy of this measure; why is it enabled by default for builds but not for installations? And what is the correct way to configure the port so that the dependency libraries are available at runtime?

**Comment by Neumann-A** [comment]
You are probably looking for the option `ADD_BIN_TO_PATH` of `vcpkg_cmake_install`

see https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_cmake_install#add_bin_to_path

**Reply by F-I-D-O** [reply]
I've just read the documentation on this option and its behavior. But I'm still confused. 

As I understand it, the `ADD_BIN_TO_PATH ` option adds the `bin` directory inside the build dir to the path. That is something different compared to the default CMake build with the vcpkg toolchain, where libraries are copied from the vcpkg local repo directly to the build dir.

How is this supposed to work?

**Reply by Neumann-A** [reply]
You don't need to copy dlls if the directory of the containing the dlls is on PATH. As such there is no need to copy them. 

> directory inside the build dir to the path.

It is the installed dir not the build dir.

**Reply by F-I-D-O** [reply]
>It is the installed dir not the build 

I see. Thank you! After careful examination, I've found out that it works. I was confused as the error was similar (missing DLLs) even after using the `ADD_BIN_TO_PATH`. However, now it only fails for google test, which can be a different issue, as I do not install it with vcpkg but instead, according to the doc, using the `FetchContent`...
